### DISCUSSION TITLE
Cannot continue printing after a pause

### DISCUSSION THREAD

**Discussion by urdigor13** [synman/Ender-3-S1-Pro-Firmware#40] [Q&A] [answered]
Unable to continue printing after g-code M25. When you press the "continue printing" button on the display, nothing happens, printing does not continue. I also tried to pause using the display. It also fails to continue printing. Tell me what could be the problem?

**Comment by synman** [comment]
I'm not sure at this point.... Will need to test it myself.

You're sending the M25 externally and then trying to resume via the screen?

Is it parking the extruder when you send it?

Have you tried sending M24 to resume?

Does it also fail if you try to pause/resume (both operations) using the touchscreen?

This code is untouched from stock but that doesn't mean Creality didn't leave us a bug.

**Comment by urdigor13** [comment]
After executing the M25 code, printing is suspended and the extruder is parked. The display shows the message "The filaments are running out. Please feed new filaments or stop printing". Two buttons "Yes" and "Stop" are displayed at the bottom. 
Clicking on "Yes" displays the following message "Filaments loaded. Continue or stop printing.", two buttons are displayed at the bottom: "Continue" and "No". 
When the "Continue" button is pressed, the screen returns to its normal state, but printing does not resume. If you cancel printing (press the stop button on the display), then the printer enters some kind of inadequate state. It does not hang, but it begins to react inadequately to commands. For example, if you try to park the extruder, the message "Auto back home, please wait..." is displayed on the display, but nothing happens and the printer freezes. 
I have to turn it off/on to restart it.
When you try to resume printing using the M24 command, nothing happens. The message "The filaments are running out" remains lit on the screen. Please feed new filaments or stop printing" and printing does not resume.

**Reply by synman** [reply]
One more question before I dig into code.... Are you able to confirm this does not happen on the stock firmware?

**Comment by urdigor13** [comment]
@synman, thank you very much for working with the firmware! And thank you for helping me! I will definitely compare the behavior of the firmware and let you know. A little bit later. The printer has urgent work for the next 12 hours :)

**Reply by synman** [reply]
I think you will find it works.

I suspect the implementation of host action commands, emergency parser, and advanced pause may be confusing the existing (stupid) Creality code.

I will likely be converting this discussion to a bug.  I just need to find that one piece of logic I can tie it to to get it going.

I also deal with a lot of "noise" when it comes to issues being reported.  My apologies if I'm making you do more ground work than you'd like to.

**Comment by urdigor13** [comment]
Good day! On stock firmware, the m25 and m24 commands work adequately:
After the m25 command, printing is suspended. The extruder stays in place, does not move anywhere. Printing can be continued either using the display or using the m24 command.

**Reply by synman** [reply]
yeah, It's advanced pause that is causing us some issues here....  that coupled with host action commands is triggering "user input" requirement.

the Creality screen handler doesn't know the difference and just assumes its the runout sensor because the extruder parked.  If you are connected via a sender that supports host action commands like octoprint you can resume just fine.  I imagine pronterface would be the same.

I'm working through how best to address this now.  Will likely have a solution (at the least a work-around) before I call it a night.

**Comment by synman** [comment, marked as answer]
See #44 for further progress

**Comment by pacocatech** [comment]
any updates? using version 2.1.x I had the same problem when I call the m600 command. When I click continue it doesn't come back.
